### PR TITLE
Feature/image equation

### PR DESCRIPTION
**Pull request by Frexor**
Adding more numbers in a row and mathematical symbols. Adding the offset to the next characters in the equation.
Sample character layout:
number

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -7,77 +7,119 @@
...
+        }
+        if (number == '+') {
+            drawLine(img, offset, false, false, false, false, false, false, false, false, true, false);
+        }
+        if (number == '=') {
+            drawLine(img, offset, false, false, false, false, false, false, false, false, false, true);
+        }
+    }
+
+    static void drawLine(BufferedImage img, int offset, boolean w1, boolean w2, boolean w3, boolean h1, boolean h2, boolean h3, boolean h4, boolean e1, boolean e2, boolean e3) throws IOException {
```
Comment (piekaa) [inline src/Main.java]: throws IOException not needed here

File: src/Main.java [hunk 2]
```diff
@@ -7,77 +7,119 @@
...
+        drawNumber(img, secondNumber, 200);
+        drawSymbol(img, secondSymbol, 300);
+        drawNumber(img, thirdNumber, 400);
+
+        File outputFile = new File("number.png");
+        ImageIO.write(img, "png", outputFile);
+
+    }
+
+    static void drawNumber(BufferedImage img, char number, int offset) throws IOException {
```
Comment (piekaa) [inline src/Main.java]: throws IOException not needed here

File: src/Main.java (hunk 1 above)
Comment (piekaa) [inline src/Main.java]: w1, w2, w3 stands for what? My idea was to call them v1, v2....., h1, h2....
v - vertical
h - horizontal

Comment (piekaa) [inline src/Main.java]: For drawing number there is always false, false, false at the end, for drawing symbols it's the other way around. Consider creating separate methods for drawing lines for numbers, and lines for symbols. I would find it a bit more readable. 
like:
```
static void drawNumberLine(BufferedImage img, int offset, boolean v1, boolean v2, boolean v3, boolean h1, boolean h2, boolean h3, boolean h4)
```
and
```
static void drawSymbolLine(BufferedImage img, int offset, boolean e1, boolean e2, boolean e3)
```
What's more drawLine for symbols now sometimes creates two lines (for + and =) thus it's lines not line.

Even better solution would be to create three methods:
`drawPlus`
`drawMinus`
`drawEqual`

and remove three last parameters from `drawLine`

Thanks to that, we'll get rid of three if statements.

File: src/Main.java [hunk 3]
```diff
@@ -7,77 +7,119 @@
 
 public class Main {
 
-    static void drawLine(BufferedImage img, boolean w1, boolean w2, boolean w3, boolean h1, boolean h2, boolean h3, boolean h4) {
+    public static void main(String[] args) throws IOException {
+        BufferedImage img = new BufferedImage(500, 200, BufferedImage.TYPE_INT_RGB);
         Graphics2D g2d = img.createGraphics();
```
Comment (piekaa) [inline src/Main.java]: Would be more clean if you extract 
```
Graphics2D g2d = img.createGraphics();
g2d.setColor(Color.WHITE);
g2d.fillRect(0, 0, img.getWidth(), img.getHeight());
g2d.dispose();
```
to separate method, let's say drawBackground, or clearToColor, or fillWithWhite, or sth like that

**Review by piekaa — CHANGES_REQUESTED** [review]
Works fine, good job. Some changes can be done to improve code quality.

File: src/Main.java (hunk 1 above)
Comment (Frexor) [reply src/Main.java]: Done.

File: src/Main.java (hunk 2 above)
Comment (Frexor) [reply src/Main.java]: Done.

File: src/Main.java (hunk 1 above)
Comment (Frexor) [reply src/Main.java]: Corrected as suggested. Instead of "w" there is "v" now, according to the English language.

Comment (Frexor) [reply src/Main.java]: Symbols have been split into `drawSymbol` to read the typed characters.
Each symbol got its own method:
`drawPlus`
`drawMinus`
`drawEqual`

File: src/Main.java (hunk 3 above)
Comment (Frexor) [reply src/Main.java]: Done, separate to `drawBackground`

File: src/Main.java [hunk 4]
```diff
@@ -7,77 +9,143 @@
...
+        }
+        if (number == '+') {
+            drawPlus(img, offset);
+        }
+        if (number == '=') {
+            drawEqual(img, offset);
+        }
+    }
+
+    static void drawLine(BufferedImage img, int offset, boolean v1, boolean v2, boolean v3, boolean h1, boolean h2, boolean h3, boolean h4, boolean e1, boolean e2, boolean e3) {
```
Comment (piekaa) [inline src/Main.java]: `boolean e1, boolean e2, boolean e3`
these are not needed anymore

**Review by piekaa — CHANGES_REQUESTED** [review]

**Review by piekaa — APPROVED** [review]
Debesta

File: src/Main.java (hunk 4 above)
Comment (Frexor) [reply src/Main.java]: Done, deleted unnecessary code.

**Pull request merged**